boards/bluepill-stm32f103cb/Makefile.include

```diff
@@ -1,3 +1,6 @@
 OPENOCD_CONFIG ?= $(RIOTBOARD)/common/blxxxpill/dist/openocd-128kib.cfg
 
+USB_VID ?= $(USB_VID_TESTING)
```

Comment:
Why is it now necessary to set `USB_{VID,PID}` to `USB_{VID,PID}_TESTING` by default?

AFAIK, `USB_{VID,PID}_TESTING` (1209:7d01) are only intended to be used for test applications. Applications that just use the USB peripheral should use the RIOT standard peripheral VID/PID 1209:7d00 if not overridden by user's VID/PID (https://pid.codes/org/RIOT).

However, setting `USB_{VID,PID}_TESTING` as default here, prevents to use the RIOT standard peripheral VID/PID 1209:7d00.

Reply:
Hmmm, I think the issue is that one must enable `KCONFIG_USB` for these boards, which then takes the default values, which sets the values and causes an error since they shouldn't be set explicitly.

Maybe it would be better to set the default kconfig value to 0x7D00 here:

```
config USB_PID
    hex "Product ID"
    range 0x0000 0xFFFF
    default 0x7D01
    help
        You must provide your own PID.
```

But that causes the makefile to fail saying you should not explicitly set it to that.

I will keep trying.  I think I still need some to do some makefile cleanup anyways as there are some TERM and TERM_DELAY setting getting resolved in the Makefile.dep that should probably only come after these things are resolved or so.

Reply:
> Maybe it would be better to set the default kconfig value to 0x7D00 here:

There is an exception, it has to be configured to 1209:7d02 for `bootloaders/booloader_dfu`, but ths doesn't have a `Kconfig`/`app.config` file yet.

Reply:
Maybe when we get to that we can have a `default x if <DFU>`.  That should be manageable.

Reply:
So just a quick update on this thread:

There is a `USB_H_USER_IS_RIOT_INTERNAL`, `INTERNAL_PERIPHERAL_*`, `USB_*ID` and `TESTING_USB_*ID` that all play together and have some sort of validation.

I will take a closer look into it to see if there is an easier way to manage it...